src/use_case/login/LoginInteractor.java

```diff
@@ -0,0 +1,56 @@
+package use_case.login;
+
+import java.util.ArrayList;
+/*
+Missing Project entity
+ */
+
+public class LoginInteractor implements LoginInputBoundary {
+    private final LoginDataAccessInterface userDataAccessObject;
+    private final LoginOutputBoundary loginPresenter;
+    private ArrayList<ArrayList<String>> projectData;
+
+    public LoginInteractor(LoginDataAccessInterface userDataAccessInterface,
+                           LoginOutputBoundary loginOutputBoundary) {
+        this.userDataAccessObject = userDataAccessInterface;
+        this.loginPresenter = loginOutputBoundary;
+    }
+
+    @Override
+    public void execute(LoginInputData loginInputData) {
+        String username = loginInputData.getUsername();
+        String password = loginInputData.getPassword();
+        if (userDataAccessObject.getUser(username) == null) {
+            loginPresenter.prepareFailView(username + ": Account does not exist.");
+        } else {
+            String pwd = userDataAccessObject.getUser(username).getPassword(); // User needs a method getPassword()
+            if (!password.equals(pwd)) {
+                loginPresenter.prepareFailView("Incorrect password for " + username + ".");
+            } else {
+
+                User user = userDataAccessObject.getUser(loginInputData.getUsername());
+
+                ArrayList<Project> listofproject = user.getProjects(); // get arraylist of projects
+                for (Project project: listofproject) {
+                    ArrayList<Task> tasks = project.getTasks(); // return arraylist of tasks for the project
+                    String stringTasks = "";
```

Comment:
stringTasks can be made a StringBuilder, since it is more efficient to append to a StringBuilder than concatenate strings.